include/oneapi/dpl/internal/common_config.h

```diff
@@ -16,12 +16,18 @@
 #ifndef _ONEDPL_COMMON_CONFIG_H
 #define _ONEDPL_COMMON_CONFIG_H
 
-#if __cplusplus < 201703L
+#ifndef _MSVC_LANG
+#    define _ONEDPL_CPLUSPLUS __cplusplus
+#else
+#    define _ONEDPL_CPLUSPLUS _MSVC_LANG
+#endif
+
+#if _ONEDPL_CPLUSPLUS < 201703L
 #    error "oneDPL requires the C++ language version not less than C++17"
 #endif
 
 // Workarounds for libstdc++9, libstdc++10 when new TBB version is found in the environment
-#if __cplusplus >= 201703L
+#if _ONEDPL_CPLUSPLUS >= 201703L
```

Comment:
Here we do not need it, as the code below is only for the GCC standard library. Probably it makes sense to add here a check for `_GLIBCXX_RELEASE` presence instead.

Reply:
I think it won't work. `_GLIBCXX_RELEASE` is not defined yet in this place so whatever check we apply `_GLIBCXX_RELEASE` is going to be zero.

Reply:
Fixed: rolled back my changes in this place.